modules/ding_user/ding_user.module

```diff
@@ -937,3 +944,17 @@ function ding_user_form_contact_site_form_alter(&$form, $form_state, $form_id) {
 function _ding_user_comment_cache_id($uid) {
   return 'ding_user_comment_user_name:' . $uid;
 }
+
+/**
+ * Check if newly created user has assigned new roles.
+ *
+ * @param array $form
+ *   Register form elements.
+ * @param array $form_state
+ *   State of the submitted form.
+ */
+function _ding_user_roles_form_validate(array $form, array &$form_state) {
+  if (!array_filter($form_state['values']['roles_change'])) {
```

Comment:
On user registration the roles array is not populated, it is on user edit. For user registration roles_change array is populated. So, only on this array can be verified if a role has been assigned.